zorch/commit/ajtai.py

```
@@ -0,0 +1,192 @@
+    return all(
+        norms.linf(rns.reconstruct_centered(host, ring.q_moduli)) <= beta_inf
+        for host in _elements_as_host(ring, batched)
+    )
+
+
+def _equal(a: Eval, b: Eval) -> bool:
+    """Exact per-limb equality, materialised to the host — verification is a
+    boundary predicate, not a traced op."""
+    return all(
+        np.array_equal(np.asarray(x).astype(object), np.asarray(y).astype(object))
+        for x, y in zip(a.limbs, b.limbs)
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)